lib/matplotlib/legend.py

```diff
@@ -553,6 +562,28 @@ def __init__(self, parent, handles, labels,
         self.set_title(title, prop=tprop)
         self._draggable = None
 
+        # set the text color
+        if textcolor is None:
+            pass
+        elif textcolor == 'linecolor':
+            for line, text in zip(self.get_lines(), self.get_texts()):
+                text.set_color(line.get_color())
+        elif textcolor == 'markerfacecolor' or textcolor == 'mfc':
+            for line, text in zip(self.get_lines(), self.get_texts()):
+                text.set_color(line.get_markerfacecolor())
+        elif textcolor == 'markeredgecolor' or textcolor == 'mec':
+            for line, text in zip(self.get_lines(), self.get_texts()):
+                text.set_color(line.get_markeredgecolor())
+        elif type(textcolor) in [list, tuple]:
```

Comment:
This needs to be a bit more careful so that RGBA tubles work correctly.

Reply:
Oh, I meant to ask about this. Can you point me towards an example of how this is done?

Reply:
I've taken a crack at this in my latest commit. Still, if there is an example of how it can be done better, I would be interested.